app/knight_class.py

```diff
@@ -0,0 +1,34 @@
+from __future__ import annotations
+
+
+class Knight:
+    def __init__(self, knight_data: dict) -> None:
+        self.name = knight_data.get("name")
+        self.power = self.calc_stat(knight_data, "power")
+        self.hp = self.calc_stat(knight_data, "hp")
+        self.protection = self.calc_stat(knight_data, "protection")
+
+    def calc_stat(self, kd: dict | list, stat: str) -> int:

```

Comment:
Do not use abbreviations or short function names without the need. Change function and arguments names